src/test/java/com/klarna/hiverunner/builder/HiveShellBaseTest.java

```diff
@@ -148,6 +151,42 @@ public void executeScriptNotStarted() throws IOException {
       shell.execute(UTF_8, Paths.get(file.toURI()));
     }
 
+    @Test
+    public void scriptFilesAreImportedInQueries() throws IOException {
+      String hql = "use default";
+
+      File importedFile = new File(tempFolder.getRoot(), "imported_script.hql");
+      Files.write(hql, importedFile, UTF_8);
+
+      HiveShell shell = createHiveShell();
+      shell.start();
+
+      String importHql = "source " + importedFile.getAbsolutePath();
+      List<String> results = shell.executeQuery(importHql);
+
+      assertThat(results, is(empty()));
+      verify(container).executeStatement(hql);
+    }
+
+    @Test
+    public void scriptFilesAreImportedInOtherScripts() throws IOException {
+      String hql = "use default";
+
+      File importedFile = new File(tempFolder.getRoot(), "imported_script.hql");
+      Files.write(hql, importedFile, UTF_8);
+
+      HiveShell shell = createHiveShell();
+      shell.start();
+
+      String importHql = "source " + importedFile.getAbsolutePath();
+      File file = new File(tempFolder.getRoot(), "script.hql");
+      Files.write(importHql, file, UTF_8);
+
+      shell.execute(file);
+
+      verify(container).executeStatement(hql);
+    }
+
     private HiveShell createHiveShell(String... keyValues) {
```

Comment:
What about a test for whatever would result in a null file path ended up in the class under test?